courses-app/src/app/auth/services/session-storage.service.ts

```diff
@@ -0,0 +1,42 @@
+import { Inject, Injectable, InjectionToken, inject } from '@angular/core';
+import {DOCUMENT} from '@angular/common';
+
+const WINDOW = new InjectionToken<Window>(
+  'An abstraction over global window object',
+  {
+      factory: () => {
+          const {defaultView} = inject(DOCUMENT);
+
+          if (!defaultView) {
```

Comment:
it would be better to use build-in isPlatformBrowser function + PLATFORM_ID , just google
here some example: https://dev-gang.ru/article/angular-provaider-obekta-window-np35rizdrz/